### PR TITLE
Add changelog for #7413 (Fix instantiating an indefinite Backpack)

### PR DESCRIPTION
**Pull request by ezyang**
Please include the following checklist in your PR:

* [X] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [X] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [X] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Review by Mikolaj — APPROVED** [review]
Ritual performed correctly.

**Review by emilypi — APPROVED** [review]

**Comment by fgaz** [comment]
@Mergifyio backport 3.6

File: changelog.d/pr-7413 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+synopsis: Fix instantiating an indefinite Backpack from Hackage with an inplace package
+packages: Cabal
```
Comment (fgaz) [inline changelog.d/pr-7413:2]:
```suggestion
packages: cabal-install
```

File: changelog.d/pr-7413 [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+prs: #7413
+issues: #6835
+description: {
+
+Previously, Cabal would always attempt to put instantiations of indefinite packages from Hackage in
```
Comment (fgaz) [inline changelog.d/pr-7413:7]:
```suggestion
Previously, cabal-install would always attempt to put instantiations of indefinite packages from Hackage in
```

**Review by fgaz — CHANGES_REQUESTED** [review]
I think this was purely a cabal-install change

**Review by fgaz — APPROVED** [review]
Thanks!

**Pull request merged**